### PR TITLE
measures characteristic

### PR DESCRIPTION
**Pull request by ddooley**
and inverse 'characteristic measured by'

### CODE REVIEW

File: src/ontology/ro-edit.owl
Comment (bpeters42) [inline src/ontology/ro-edit.owl:1]: - Diff has a bunch of lines moving around. More than expected. 
- Inverse property is incorrect,.

**Review by bpeters42 — CHANGES_REQUESTED** [review]
Diff has a bunch of lines moving around. More than expected.
Inverse property is incorrect,.

File: src/ontology/ro-edit.owl
Comment (ddooley) [reply src/ontology/ro-edit.owl:1]: Fixed Inverse of 'measures characteristic'
Added inverse of axiom.

**Comment by wdduncan** [comment]
Since the the relation holds between a process and a quality, do we to make the label more reflective of that? E.g.:
- `process measures characteristic`
- `characteristic measured by process`

This would make it clear that it is not a datum that holds measurement information.

**Comment by wdduncan** [comment]
Since the COB import has been added to the ODK file, I think this addresses #716.   
Do others agree?

**Comment by ddooley** [comment]
Re "process measures characteristic" label etcl, I feel it is intuitive that "measures" is tied to a measuring process, so no need to spell that out in label.

**Comment by wdduncan** [comment]
I agree 'measures' brings to mind a process, but I thought it may need to be made clearer what is doing the measuring. 
E.g., When I step on scale, there is a measurement that happens. But, just reading the label `measures characteristic`, one might not understand whether it is it the scale (i.e., device) that measures my weight or the applying of pressure (i.e., process) that does the measuring?  
For the `measures characteristic` relation, it is the latter, but I can see how someone might think it is the former (i.e., device).

**Comment by bpeters42** [comment]
I added my comments to the issue rather than this pull request so that they can viewed in context.

**Comment by ddooley** [comment]
I'm going to have to redo this one.  Its getting too complicated trying to resolve diffs with pre-UBERON integration

**Comment by anitacaron** [comment]
Hi @ddooley, I'll update this PR to align with the master. You don't need to worry about redoing the work.

**Review by wdduncan — APPROVED** [review]

**Comment by anitacaron** [comment]
Note: COB import is empty. If there isn't a COB term to import, I'd recommend removing the COB import.

**Comment by ddooley** [comment]
Yes I guess that COB import can be removed. I'd put it in thinking I could import OBI terms that way, but to my surprise one can't fetch 3rd party terms from COB indirectly (as mentioned in last RO meeting).

**Comment by anitacaron** [comment]
There's another issue to address the COB import #716. It will be added when needed.

**Comment by wdduncan** [comment]
Can you leave the COB import? I need it for one of my PRs. This will keep me from having to recreate it.

**Comment by anitacaron** [comment]
I'll add it back, fixing issue #716.

**Comment by wdduncan** [comment]
Thanks @anitacaron

**Comment by ddooley** [comment]
The range was meant to be COB characteristic, but for now that translates to PATO:00000001 quality.

As for domain, @bpeters42 I recall you suggested "assay" for domain since that was doing the measuring, and I recall assay can't measure nonmaterial things like rates directly, but our solution is that assay can measure material things that are proxies for things like rates of change.  So I've changed label to "assay" on that basis. Unfortunately it added some superfluous diffs too tho.

**Comment by ddooley** [comment]
@anitacaron want me to just do a redo on this one?  Might be simpler.

**Comment by anitacaron** [comment]
@ddooley you can change the range to BFO:0000020 now.

File: src/ontology/ro-edit.owl
Comment (ddooley) [reply src/ontology/ro-edit.owl:1]: I think @anitacaron fixed most of the diffs by updating this branch with recent master.

**Comment by ddooley** [comment]
Switch to BFO:0000020 is now done!

**Comment by ddooley** [comment]
I think @bpeters42 conversation about big diff is also taken care of with your work.

**Comment by ddooley** [comment]
I did another merge of master into this branch and I think outstanding issues are taken care of?

File: src/ontology/ro-edit.owl [hunk 1]
```diff
@@ -5988,6 +5990,25 @@ AnnotationAssertion(rdfs:label obo:RO_0009005 "has primary substance added"@en)
 SubObjectPropertyOf(obo:RO_0009005 obo:RO_0009001)
 IrreflexiveObjectProperty(obo:RO_0009005)
 
+# Object Property: obo:RO_0009006 (assay measures characteristic)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0009006 "A relation between an assay and a characteristic, in which the assay generates a data item which is a measure of a characteristic."@en)
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:5995]: Could you add an example of usage, please?

File: src/ontology/ro-edit.owl [hunk 2]
```diff
@@ -5988,6 +5990,25 @@ AnnotationAssertion(rdfs:label obo:RO_0009005 "has primary substance added"@en)
...
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0009006 <https://orcid.org/0000-0002-8844-9165>)
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:5996]: Could you please change the annotation to `terms:contributor`?

File: src/ontology/ro-edit.owl [hunk 3]
```diff
@@ -5988,6 +5990,25 @@ AnnotationAssertion(rdfs:label obo:RO_0009005 "has primary substance added"@en)
...
+AnnotationAssertion(terms:date obo:RO_0009006 "2023-05-23T15:24:15Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:label obo:RO_0009006 "assay measures characteristic"@en)
+InverseObjectProperties(obo:RO_0009006 obo:RO_0009007)
+ObjectPropertyDomain(obo:RO_0009006 obo:OBI_0000070)
+ObjectPropertyRange(obo:RO_0009006 obo:BFO_0000020)
+
+# Object Property: obo:RO_0009007 (characteristic measured by assay)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0009007 "Inverse of 'measures characteristic'"@en)
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0009007 <https://orcid.org/0000-0002-8844-9165>)
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:6006]: Could you please change the annotation to `terms:contributor`?

**Review by anitacaron — CHANGES_REQUESTED** [review]

File: src/ontology/ro-edit.owl (hunk 1 above)
Comment (ddooley) [reply src/ontology/ro-edit.owl]: Done!

File: src/ontology/ro-edit.owl (hunk 3 above)
Comment (ddooley) [reply src/ontology/ro-edit.owl]: done!

**Comment by ddooley** [comment]
@wdduncan @bpeters42 final touchups are finished!

**Review by anitacaron — APPROVED** [review]
Technically, I approve. It still needs an ontology review.

File: src/ontology/ro-edit.owl (hunk 3 above)
Comment (anitacaron) [reply src/ontology/ro-edit.owl]: Thanks!

**Review by wdduncan — APPROVED** [review]

File: src/ontology/ro-edit.owl [hunk 4]
```diff
@@ -6002,6 +6004,26 @@ AnnotationAssertion(rdfs:label obo:RO_0009005 "has primary substance added"@en)
...
+AnnotationAssertion(terms:contributor obo:RO_0009006 <https://orcid.org/0000-0002-8844-9165>)
+AnnotationAssertion(terms:date obo:RO_0009006 "2023-05-23T15:24:15Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:label obo:RO_0009006 "assay measures characteristic"@en)
+InverseObjectProperties(obo:RO_0009006 obo:RO_0009007)
+ObjectPropertyDomain(obo:RO_0009006 obo:OBI_0000070)
+ObjectPropertyRange(obo:RO_0009006 obo:BFO_0000020)
+
+# Object Property: obo:RO_0009007 (characteristic measured by assay)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0009007 "Inverse of 'assayt measures characteristic'"@en)
```
Comment (bpeters42) [inline src/ontology/ro-edit.owl:6020]: typo: assayt --> assay

**Comment by bpeters42** [comment]
I think the pull request as is doesn't do much harm; there is still a lot to be resolved for when we transition to modeling values of measurements via @jamesaoverton approach, but this will not make that worse, and it has been requested for a while. So I would approve. (apart from the minor typo I found)

**Comment by jamesaoverton** [comment]
@bpeters42 I have concerns about the bigger picture here, but I agree that this PR should go ahead.

File: src/ontology/ro-edit.owl (hunk 4 above)
Comment (anitacaron) [reply src/ontology/ro-edit.owl]: done

**Review by anitacaron — APPROVED** [review]
Approving again after fixing a typo in the inverse property's definition.

**Review by wdduncan — APPROVED** [review]

**Pull request merged**